worlds/messenger/__init__.py

```diff
@@ -54,58 +67,150 @@ class MessengerWorld(World):
+        },
+        "Phobe": {
+            "Catacombs - Necro",
+            "Bamboo Creek - Claustro",
+            "Searing Crags - Pyro",
+            "Cloud Ruins - Acro",
+        },
+        "Phobekin": {
+            "Catacombs - Necro",
+            "Bamboo Creek - Claustro",
+            "Searing Crags - Pyro",
+            "Cloud Ruins - Acro",
+        },
+    }
 
-    required_client_version = (0, 4, 2)
+    required_client_version = (0, 4, 3)
 
     web = MessengerWeb()
 
     total_seals: int = 0
     required_seals: int = 0
+    created_seals: int = 0
     total_shards: int = 0
     shop_prices: Dict[str, int]
     figurine_prices: Dict[str, int]
     _filler_items: List[str]
+    starting_portals: List[str]
+    plando_portals: List[str]
+    spoiler_portal_mapping: Dict[str, str]
+    portal_mapping: List[int]
+    transitions: List[Entrance]
+    reachable_locs: int = 0
 
     def generate_early(self) -> None:
         if self.options.goal == Goal.option_power_seal_hunt:
-            self.options.shuffle_seals.value = PowerSeals.option_true
             self.total_seals = self.options.total_seals.value
 
+        if self.options.limited_movement:
+            self.options.accessibility.value = Accessibility.option_minimal
+            if self.options.logic_level < Logic.option_hard:
+                self.options.logic_level.value = Logic.option_hard
+
+        if self.options.early_meditation:
+            self.multiworld.early_items[self.player]["Meditation"] = 1
+
         self.shop_prices, self.figurine_prices = shuffle_shop_prices(self)
 
+        starting_portals = ["Autumn Hills", "Howling Grotto", "Glacial Peak", "Riviere Turquoise", "Sunken Shrine", "Searing Crags"]
+        self.starting_portals = [f"{portal} Portal"
+                                 for portal in starting_portals[:3] +
+                                 self.random.sample(starting_portals[3:], k=self.options.available_portals - 3)]
+        # super complicated method for adding searing crags to starting portals if it wasn't chosen
+        # need to add a check for transition shuffle when that gets added back in
+        if not self.options.shuffle_portals and "Searing Crags Portal" not in self.starting_portals:
+            self.starting_portals.append("Searing Crags Portal")
+            if len(self.starting_portals) > 4:
+                portals_to_strip = [portal for portal in ["Riviere Turquoise Portal", "Sunken Shrine Portal"]
+                                    if portal in self.starting_portals]
+                self.starting_portals.remove(self.random.choice(portals_to_strip))
+
+        self.plando_portals = []
+        self.portal_mapping = []
+        self.spoiler_portal_mapping = {}
+        self.transitions = []
+
+    @classmethod
+    def stage_generate_early(cls, multiworld: MultiWorld):
+        if multiworld.players > 1:
+            return
+        cls.generate_output = generate_output
```

Comment:
yeah i put this here because i didn't like that it said "generating output for 20 players" and then didn't actually output any files. thought i had removed it though. it's moved to a classmethod now.